packages/python-google-compute-engine/google_compute_engine/networking/network_daemon.py

```diff
@@ -119,7 +122,9 @@ def _ExtractInterfaceMetadata(self, metadata):
         if self.target_instance_ips:
           ip_addresses.extend(network_interface.get('targetInstanceIps', []))
         interfaces.append(NetworkDaemon.NetworkInterface(
-            interface, ip_addresses, network_interface.get('ip', [])))
+            interface, forwarded_ips=ip_addresses,
+            ip=network_interface.get('ip', None),
```

Comment:
why change [] to None ?

Reply:
It's not changing. This is saying retrieve `ip` and if it not set, return None. The `NetworkInterface` object has None as the default rather than an empty list.